src/jobflow_remote/cli/formatting.py

```diff
@@ -579,6 +579,169 @@ def get_flow_report_components(report: FlowsReport) -> list[RenderableType]:
     return components
 
 
+def get_single_flow_report_components(flow_info: FlowInfo) -> list[RenderableType]:
+    """
+    Generate report rich components for a single Flow.
+
+    Parameters
+    ----------
+    flow_info
+        The FlowInfo object containing information about the Flow.
+
+    Returns
+    -------
+    list[RenderableType]
+        List of Rich components for display.
+    """
+    components = []
+
+    # Flow Header with basic information
+    header_table = Table(
+        title=f"Flow Report: {flow_info.name}",
+        title_style="bold green",
+        show_header=False,
+    )
+    header_table.add_column("", style="cyan", justify="right")
+    header_table.add_column("", style="white", justify="left")
+
+    header_table.add_row("Flow ID", flow_info.flow_id)
+    header_table.add_row("Flow State", flow_info.state.name)
+    header_table.add_row("Number of Jobs", str(len(flow_info.job_ids)))
+
+    # Calculate timing information
+    start_time = flow_info.created_on
+    header_table.add_row(
+        "Start Time",
+        convert_utc_time(start_time).strftime(fmt_datetime) + f" [{time.tzname[0]}]",
+    )
+
+    # If flow is completed, calculate total time
+    if flow_info.state == FlowState.COMPLETED:
+        end_time = flow_info.updated_on
+        header_table.add_row(
+            "Complete Time",
+            convert_utc_time(end_time).strftime(fmt_datetime) + f" [{time.tzname[0]}]",
+        )
+        total_time = (end_time - start_time).total_seconds()
+        hours, remainder = divmod(total_time, 3600)
```

Comment:
Maybe make a utility function ?

Reply:
you mean here, or a generic one in the `util` module?